DatadogRUM/Sources/RUMMonitor/Scopes/FatalErrorContextNotifier.swift

```diff
@@ -7,9 +7,22 @@
 import Foundation
 import DatadogInternal
 
-/// Manages RUM information necessary for building context for fatal errors such as Crashes or Fatal App Hangs.
+/// A type managing RUM information necessary for building context of fatal errors such as Crashes or Fatal App Hangs.
+internal protocol FatalErrorContextNotifying: AnyObject {
+    /// The state of the current RUM session.
+    /// Can be `nil` if no session was yet started. Never gets `nil` after starting first session.
+    var sessionState: RUMSessionState? { set get }
+    /// The active RUM view in current session.
+    /// Can be `nil` if no view is yet started. Will become `nil` if view was stopped without starting the new one.
+    var view: RUMViewEvent? { set get }
+    /// The current set of RUM attributes.
+    /// It gets updated with global attributes being added or removed in `RUMMonitor`.
+    var globalAttributes: [String: Encodable] { set get }
+}
+
+/// Manages RUM information necessary for building context of fatal errors such as Crashes or Fatal App Hangs.
 /// It tracks value changes and notifies updates on message bus.
-internal final class FatalErrorContextNotifier {
+internal final class FatalErrorContextNotifier: FatalErrorContextNotifying {
```

Comment:
/suggestion I think the names should reflect responsibility (interface) vs. behavior (imp). Here it would sending the context vs. sending on the bus.
```swift
internal final class FatalErrorContextBus: FatalErrorContextSender
```

Reply:
It's true for strategy pattern, but is not the case here. The protocol is made purely for dependency injection and testability. It is not for introducing multiple strategies of sending `FatalErrorContext`. 

If at any point we want to notify FEC through different implementations, then I'll be 100% on the side of this suggestion. For now I'll keep it as it is 🙂.